packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1091,6 +1105,28 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
 
       final double xOffset = hasCustomWidth ? (size.width - snackBarWidth!) / 2 : 0.0;
       positionChild(_ScaffoldSlot.snackBar, Offset(xOffset, snackBarYOffsetBase - snackBarSize.height));
+
+      assert((){
+        // Whether a SnackBar has been offsetted too high by floatingActionButton.
+        // Assert after call to positionChild otherwise SnackBar position differs in debug mode.
+        if (floatingActionButtonRect.size != Size.zero && isSnackBarFloating) {
```

Comment:
This `if` statement already exists above on line 1092, can this just be included in that block?

Reply:
Perhaps not, since the position of a floating snackbar can be affected by other Scaffold widgets too, like persistent footer buttons and bottom navigation bars, can you account for these cases too?

```suggestion
        if ( isSnackBarFloating) {
```

Maybe? If there is no floating action button, how are the other cases handled?

Reply:
@Piinks 
Very interesting! Large persistent footer buttons and large bottom navigation bars might happen too. I updated the PR accordingly.
About the if statement location, the check has to be done after calling positionChild otherwise the error is thrown before position is calculated and the SnackBar position will differ in debug mode (I first put the message in the existing if statement and noticed that).